repositories/cyclonedds.BUILD.bazel

```diff
@@ -146,11 +154,10 @@ cmake(
     lib_source = ":all_srcs",
     linkopts = select(
         {
-            "@platforms//os:linux": ["-lpthread"],
-            "@platforms//os:macos": ["-lpthread"],
-            "@platforms//os:qnx": [],
+            ":linux_or_macos": ["-lpthread"],
+            ":qnx_or_android": [],
         },
-        no_match_error = "Only Linux, macOS and QNX are supported!",
+        no_match_error = "Only Linux, macOS , QNX and Android are supported!",
```

Comment:
Nit: let's keep the list alph-sorted (there are 4 items already): `Supported OSs: Android, Linux, macOS, QNX`. Please fix here and elsewhere.